src/Symfony/Component/Console/CommandLoader/ContainerCommandLoader.php

```diff
@@ -0,0 +1,61 @@
+<?php
+
+namespace Symfony\Component\Console\CommandLoader;
+
+use Psr\Container\ContainerInterface;
+use Symfony\Component\Console\Exception\CommandNotFoundException;
+
+/**
+ * Loads commands from a PSR-11 container.
+ *
+ * @author Robin Chalas <robin.chalas@gmail.com>
+ */
+class ContainerCommandLoader implements CommandLoaderInterface
+{
+    private $container;
+    private $commandNames;
+
+    public function __construct(ContainerInterface $container, array $commandNames)
+    {
+        $this->container = $container;
+        $this->commandNames = $commandNames;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function get($name)
+    {
+        if (!in_array($name, $this->commandNames, true)) {
```

Comment:
Could we replace this by `if(!$this->has($name))` to avoid duplicating code? (unless this function call makes a big difference in performance)